dataland-frontend/src/components/pages/SearchCompaniesForFrameworkData.vue

```diff
@@ -78,41 +109,96 @@ export default defineComponent({
 
   created() {
     window.addEventListener("scroll", this.handleScroll);
-    this.scanQueryParams();
+    this.scanQueryParams(this.route);
   },
   data() {
     return {
       searchBarToggled: false,
       pageScrolled: false,
       route: useRoute(),
-      showSearchResultsTable: false,
       resultsArray: [] as Array<DataSearchStoredCompany>,
       latestScrollPosition: 0,
       currentSearchBarInput: "",
-      currentFilteredFrameworks: [] as Array<DataTypeEnum>,
+      currentFilteredFrameworks: Object.values(DataTypeEnum) as Array<DataTypeEnum>,
+      currentFilteredCountryCodes: [] as Array<string>,
+      currentFilteredSectors: [] as Array<string>,
+      currentCombinedFilter: {
+        companyNameFilter: "",
+        frameworkFilter: [],
+        sectorFilter: [],
+        countryCodeFilter: [],
+      } as FrameworkDataSearchFilterInterface,
       scrollEmittedByToggleSearchBar: false,
       hiddenSearchBarHeight: 0,
       searchBarName: "search_bar_top",
+      indexOfFirstShownRow: 0,
+      rowsPerPage: 100,
+      waitingForSearchResults: true,
     };
   },
+  beforeRouteUpdate(to: RouteLocationNormalizedLoaded) {
+    this.scanQueryParams(to);
+  },
   setup() {
     return {
+      searchBarAndFiltersContainer: ref(),
       frameworkDataSearchBar: ref(),
-      searchBarContainer: ref(),
+      frameworkDataSearchFilters: ref(),
+      searchResults: ref(),
     };
   },
   watch: {
     pageScrolled(pageScrolledNew) {
       if (pageScrolledNew) {
-        this.frameworkDataSearchBar.$refs.autocomplete.hideOverlay();
+        this.frameworkDataSearchBar?.$refs.autocomplete.hideOverlay();
       }
       if (!pageScrolledNew) {
         this.searchBarToggled = false;
       }
     },
+    currentFilteredFrameworks: {
+      handler() {
+        this.updateCombinedFilterIfRequired();
+      },
+      deep: true,
+    },
+    currentFilteredCountryCodes: {
+      handler() {
+        this.updateCombinedFilterIfRequired();
+      },
+      deep: true,
+    },
+    currentFilteredSectors: {
+      handler() {
+        this.updateCombinedFilterIfRequired();
+      },
+      deep: true,
+    },
   },
+  computed: {
+    currentlyVisiblePageText(): string {
+      const totalSearchResults = this.resultsArray.length;
 
+      if (!this.waitingForSearchResults) {
+        if (totalSearchResults === 0) {
+          return "No results";
+        } else {
+          const startIndex = this.indexOfFirstShownRow;
+          const endIndex =
+            startIndex + (this.rowsPerPage - 1) >= totalSearchResults
+              ? totalSearchResults - 1
+              : startIndex + (this.rowsPerPage - 1);
+          return `${startIndex + 1}-${endIndex + 1} of ${totalSearchResults} results`;
+        }
+      } else {
+        return "loading...";
+      }
+    },
```

Comment:
broken:
This does not adjust correctly if you run a new query while being on a paginator page >1 of your originally query results

Reply:
Fixed